rosbag2_transport/src/rosbag2_transport/recorder.cpp

```diff
@@ -184,7 +186,9 @@ Recorder::create_subscription(
 
 void Recorder::record_messages() const
 {
-  spin(node_);
+  rclcpp::executors::MultiThreadedExecutor exe;
+  exe.add_node(node_->get_node_base_interface());
+  exe.spin();
```

Comment:
So while this allows me to use the traditional `lifecycle` command line argument, this will not work during playback time. Playback as well has to have a `spin` thread where the lifecycle services are executed.

Reply:
Are you saying we should add a spin thread to `player.cpp`, maybe inside `play()`?
I actually added the spin above for `recorder.cpp` because the test wouldn't pass without it. Nothing happens.

On a related note, as I mentioned [here](https://github.com/ros2/rosbag2/pull/443#discussion_r445903866), I actually didn't get anything with `ros2 lifecycle nodes` on the command line, same for `ros2 bag record`. Is there something else needed for that to work?